lib/bumper.js

```diff
@@ -12,14 +21,48 @@ const utils = require('./utils')
  * @class
  */
 class Bumper {
+  // ====================================================================================
+  // Public Methods
+  // ====================================================================================
 
   /**
-   * @param {Vcs} vcs - the vcs instance to use
-   * @param {Config} config - the bumper config object
+   * @param {Object} params - params obj
+   * @param {Config} params.config - the bumper config object
+   * @param {Vcs} params.vcs - the vcs instance to use
+   * @param {Ci} params.ci - the ci instance to use
    */
-  constructor (vcs, config) {
-    this.vcs = vcs
-    this.config = config
+  constructor (params) {
+    this.config = params.config
+    this.vcs = params.vcs
+    this.ci = params.ci
+  }
+
+  /**
+   * Bump the version based on the last merged PR's version-bump comment
+   * @returns {Promise} a promise resolved with the results of the push
+   */
+  bump () {
+    if (this.config.isPr) {
+      logger.log(`${pkgJson.name}: Not a merge build, skipping bump`)
+      return Promise.resolve()
+    }
+
+    return this._getMergedPrInfo()
+      .then((info) => {
+        return this._bumpVersion(info, 'package.json')
+      })
+      .then((info) => {
+        return this._prependChangelog(info, 'CHANGELOG.md')
+      })
+      .then(() => {
+        return this._commitChanges()
+      })
+      .then(() => {
+        return this._createTag()
+      })
+      .then(() => {
+        return this.ci.push(this.vcs)
+      })
```

Comment:
Could be reduced to:
```js
    return this._getMergedPrInfo()
      .then((info) => {
        return this._bumpVersion(info, 'package.json')
      })
      .then((info) => {
        return this._prependChangelog(info, 'CHANGELOG.md')
      })
      .then(this._commitChanges.bind(this))
      .then(this._createTag.bind(this))
      .then(this.ci.push.bind(this, this.vcs))
```
and if you change the order of the arguments on `_bumpVersion()` and `_prependChangelog()` it could be reduced further to:
```js
    return this._getMergedPrInfo()
      .then(this._bumpVersion.bind(this, 'package.json'))
      .then(this._prependChangelog.bind(this, 'CHANGELOG.md'))
      .then(this._commitChanges.bind(this))
      .then(this._createTag.bind(this))
      .then(this.ci.push.bind(this, this.vcs))
```